src/canmatrix/cmcsv.py

```diff
@@ -270,3 +261,7 @@ def dump(db, thefile, delimiter=',', **options):
 #        finalTableString = "\n".join(
 #            [row.toCSV(delimiter) for row in csvtable])
 #        print(finalTableString)
+    if sys.version_info > (3, 0):
+        # When TextIOWrapper is garbage collected, it closes the raw stream
+        # unless the raw stream is detached first
+        temp.detach()
```

Comment:
Funny thing, I just ran into this over in a tqdm PR.  Sensible addition given the `if` above but the thing overall sort of smells to me.  `*hopes he wasn't the one that added it*`

Should probably really be in a finally clause though just in case there are exceptions.  Or a context manager, whatever you feel like.

Reply:
I simply copied it from another module because there were an error/warning after running the Unit Test informing about unreleased resource. To be honest I don't know why the `TextIOWrapper` must be present. Either let it be for further cleanup (I can create a ticket) or I need more time to investigate. The first solution is currently more convenient for me :-)

Reply:
Yeah, all I mean for this is to make a try/finally to be sure the detach actually gets run even in case of exceptions.